### PR TITLE
Fix credentials leakage

### PR DESCRIPTION
**Pull request by pmsfc**
In this PR I've changed the exception handling to output the errors but clean the string using a regex like: https://regex101.com/r/PubRuN/2/

### CODE REVIEW

**Comment by MrKiplin** [comment]
Will need to bump the package version up in order for NPM to publish the changes. Apart from that looks good good to me 👍

File: packages/schema-registry-api/src/api.ts [hunk 1]
```diff
@@ -32,15 +32,17 @@ const apiFetch = async <T>(req: string | Request, init: RequestInit = {}): Promi
...
   let data;
   try {
+    res = await fetch(req, { ...defaultInit, ...init });
     data = await res.json();
   } catch (error) {
-    // Generic error to avoid including URL in message, as it may contain
-    // sensitive credentials.
-    throw TypeError('Schema registry responded with invalid JSON.');
+    // Use regex to sanitize error logging in order avoid basic auth credentials leakage
+    throw TypeError(error.toString().replace(/[^/]*@/g, '{CREDENTIALS}'));
```
Comment (samjacobclift) [inline packages/schema-registry-api/src/api.ts:43]: Does it make sense to just return an `Error` here rather than a `TypeError`?

**Review by samjacobclift — APPROVED** [review]
This looks fine now, as @MrKiplin said just need to bump the package version [here](https://github.com/ovotech/bit-node-tools/blob/master/packages/schema-registry-api/package.json#L4)

File: packages/schema-registry-api/src/api.ts (hunk 1 above)
Comment (pmsfc) [reply packages/schema-registry-api/src/api.ts]: Yeah makes sense, I've changed it 😉

File: packages/schema-registry-api/src/api.ts [hunk 2]
```diff
@@ -32,15 +32,17 @@ const apiFetch = async <T>(req: string | Request, init: RequestInit = {}): Promi
...
   let data;
   try {
+    res = await fetch(req, { ...defaultInit, ...init });
     data = await res.json();
   } catch (error) {
-    // Generic error to avoid including URL in message, as it may contain
-    // sensitive credentials.
-    throw TypeError('Schema registry responded with invalid JSON.');
+    // Use regex to sanitize error logging in order avoid basic auth credentials leakage
+    throw Error(error.toString().replace(/[^/]*@/g, '{CREDENTIALS}'));
```
Comment (samjacobclift) [inline packages/schema-registry-api/src/api.ts:43]: I believe some of the tests now need updating

Comment (samjacobclift) [reply packages/schema-registry-api/src/api.ts:43]: This [one](https://github.com/ovotech/bit-node-tools/blob/master/packages/schema-registry-api/test/integration.spec.ts#L116) at least

Comment (pmsfc) [reply packages/schema-registry-api/src/api.ts:43]: Just changed it! Thanks! 🙂

**Pull request merged**